packages/client/src/monaco-language-client.ts

```diff
@@ -39,6 +66,57 @@ export class MonacoLanguageClient extends BaseLanguageClient {
     protected getLocale(): string {
         return navigator.language || 'en-US'
     }
+
+    protected override registerBuiltinFeatures() {
+        this.registerFeature(new DidOpenTextDocumentFeature(this, this['_syncedDocuments']));
+        this.registerFeature(new DidChangeTextDocumentFeature(this));
+        this.registerFeature(new DidCloseTextDocumentFeature(this, this['_syncedDocuments']));
+        this.registerFeature(new CompletionItemFeature(this));
+        this.registerFeature(new HoverFeature(this));
+        this.registerFeature(new SignatureHelpFeature(this));
+        this.registerFeature(new DefinitionFeature(this));
+        this.registerFeature(new ReferencesFeature(this));
+        this.registerFeature(new DocumentHighlightFeature(this));
+        this.registerFeature(new DocumentSymbolFeature(this));
+        this.registerFeature(new CodeActionFeature(this));
+        this.registerFeature(new CodeLensFeature(this));
+        this.registerFeature(new DocumentFormattingFeature(this));
+        this.registerFeature(new DocumentRangeFormattingFeature(this));
+        this.registerFeature(new DocumentOnTypeFormattingFeature(this));
+        this.registerFeature(new RenameFeature(this));
+        this.registerFeature(new DocumentLinkFeature(this));
+        this.registerFeature(new ExecuteCommandFeature(this));
+        this.registerFeature(new TypeDefinitionFeature(this));
+        this.registerFeature(new ImplementationFeature(this));
+        this.registerFeature(new ColorProviderFeature(this));
+        // We only register the workspace folder feature if the client is not locked
+        // to a specific workspace folder.
+        if (this.clientOptions.workspaceFolder === undefined) {
+            this.registerFeature(new WorkspaceFoldersFeature(this));
+        }
+        this.registerFeature(new FoldingRangeFeature(this));
+        this.registerFeature(new DeclarationFeature(this));
+        this.registerFeature(new SelectionRangeFeature(this));
+        this.registerFeature(new SemanticTokensFeature(this));
+        this.registerFeature(new LinkedEditingFeature(this));
+        this.registerFeature(new InlayHintsFeature(this));
+        this.registerFeature(new DiagnosticFeature(this));
+    }
+
+    public registerTextDocumentSaveFeatures() {
```

Comment:
Who is responsible for calling three methods (`registerTextDocumentSaveFeatures`, `registerConfigurationFeatures`, `registerProgressFeatures`)?

Reply:
The user who is using the MonacoLanguageClient, aka,
```typescript
const languageClient = new MonacoLanguageClient(...)
languageClient.registerTextDocumentSaveFeatures()
```

Since these feature are builtin in vscode but NOT builtin in monaco (they require some additional code), I don't thing they belong in the `registerBuiltinFeatures` method

Reply:
Got it, thank you.